pkg/apis/coordination/types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package coordination
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Lease defines a lease concept.
+type Lease struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+
+	// Specification of the Lease.
+	// +optional
+	Spec LeaseSpec
+}
+
+// LeaseSpec is a specification of a Lease.
+type LeaseSpec struct {
+	// holderIdentity contains the identity of the holder of a current lease.
+	// +optional
+	HolderIdentity string
+	// leaseDurationSeconds is a duration that candidates for a lease need
+	// to wait to force acquire it. This is measure against time of last
+	// observed RenewTime.
+	// +optional
+	LeaseDurationSeconds int32
```

Comment:
Why require a specific unit (seconds) in the field name, instead of just doing `LeaseDuration metav1.Duration` and leaving the units up to the user (e.g. "10s" or "1h", etc.)? The other time-related fields in this API are unit-agnostic.

Reply:
we haven't generally used metav1.Duration for served APIs since that requires clients to implement go-compatible duration parsing. seconds are easier for clients written in arbitrary languages to deal with.

as long as we don't expect sub-second leases, I'd stick with `LeaseDurationSeconds int32`

Reply:
@liggitt is correct.

We don't use Duration in the API.

Reply:
We should update the conventions:
https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#units

Could someone please do that?

Reply:
>we haven't generally used metav1.Duration for served APIs since that requires clients to implement go-compatible duration parsing. seconds are easier for clients written in arbitrary languages to deal with.

That's a fair point - though I do think the Quantity format is more human-readable/better UX from the standpoint of looking at a config. In an ideal world generated clients would get you duration parsing for free...

Since we currently use `.*Seconds` for other served APIs, agree we should stick with that here.

Reply:
Thanks @bgrant0607 - https://github.com/kubernetes/community/pull/2248 is out for review to update conventions.